content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -1993,6 +2001,9 @@ else if(copyAction instanceof ServiceLevelAction)
 				InputStream contentStream = resource.streamContent();
 				String contentType = resource.getContentType();
 				String filename = Validator.getFileName(itemId);
+				if (!Validator.checkFilename(filename)) {
+					throw new InvalidFilenameException(filename);
```

Comment:
Same pattern :)

Reply:
Yeah, agree, the pattern is kinda bad, going to replace it.